codex-rs/linux-sandbox/src/landlock.rs

```diff
@@ -84,6 +91,34 @@ fn install_filesystem_landlock_rules_on_current_thread(
     Ok(())
 }
 
+fn gpu_device_paths() -> Vec<PathBuf> {
+    let mut paths = Vec::new();
+
+    let drm_path = Path::new("/dev/dri");
+    if drm_path.exists() {
+        paths.push(drm_path.to_path_buf());
+    }
+
+    let amd_kfd_path = Path::new("/dev/kfd");
+    if amd_kfd_path.exists() {
+        paths.push(amd_kfd_path.to_path_buf());
+    }
+
+    if let Ok(entries) = fs::read_dir("/dev") {
+        paths.extend(entries.flatten().filter_map(|entry| {
+            let file_name = entry.file_name();
+            let name = file_name.to_str()?;
+            if name.starts_with("nvidia") {
+                Some(entry.path())
+            } else {
+                None
+            }
+        }));
+    }
```

Comment:
Mali (Arm) GPU devices would typically be named 'mali?' (eg mali0) and Adreno (Qcom) devices would I think be named something along the lines of 'kgsl-*' (eg kgsl-3d0).